apps/meteor/client/lib/chats/uploads.ts

```diff
@@ -30,7 +30,7 @@ const wipeFailedOnes = (): void => {
 };
 
 const send = async (
-	file: File,
+	file: File[] | File,
```

Comment:
This API should continue sending only a single file a time. The places where we send multiple files should call this API multiple times

Reply:
Thank you for the clarification @rodrigok 

I have currently implemented it so that if there's a single file, it's converted to an array and sent using the API. For multiple files, I'm looping through each file and sending them one at a time using the API.

Reply:
@abhipatel0211 got it, but that is what is wrong. This API should only make the file upload. Should not even receive the message content to call the sendMessage inside it. The responsibility of this API should be to upload the file and return the file data to be stored in memory in the message composer. There you will call the sendMessage passing the list of files to be confirmed.

This code was like that because the flow to upload the file was coming from the modal and the old API was doing both things (upload and message send) together.

Reply:
Think that you don't need to upload the files only when the message is sent. You can upload each file when you are attaching it to the message box, so when you send the message later the files are already uploaded and you only confirm the uploads via ID. If the user forget the message in composing status the files will end up deleted after a while because they were not confirmed.

Reply:
@rodrigok sir so is it good to create an another uploadsFiles function which will be just used for the the fileUpload and return id. As this uploads function can be also used from uploadfiles and other places also (like live audio and video is sharing).

Reply:
Yes, you should extract the upload logic to another function